File: 71-a-way-too-long-words/long-words.py
w = int(input())

words = []


def abbreviate_word(word):
    word_length = len(word)

    if  word_length <= 10:
        return word

    return word[0] + str(word_length - 2) + word[-1]


for _ in range(w):
    words.append(abbreviate_word(input()))

for word in words:
    print(word)

    

File: 71-a-way-too-long-words/long-words.cpp
#include <iostream>
#include <vector>

using namespace std;

int main()
{
    int w;
    cin >> w;

    vector<string> words;

    string word;
    while (w--)
    {
        cin >> word;
        words.push_back(word);
    }

    for (auto const &word : words)
    {
        if (word.size() > 10)
        {
            cout << word[0] << word.size() - 2 << word[word.size() - 1] << endl;
        }
        else
        {
            cout << word << endl;
        }
    }
}